### PR TITLE
Typos, spelling, grammar and other small readability fixes

### PR DESCRIPTION
**Pull request by herebebeasties**

### CODE REVIEW

**Comment by stackedsax** [comment]
Thanks for the improvements!

They look good to me, but I'll let Enrico check through them.

**Comment by herebebeasties** [comment]
(@EnricoMi Should I raise a PR against the 2.4 branch too, or do you periodically merge things back anyway?)

File: README.md [hunk 1]
```diff
@@ -133,7 +133,7 @@ Dgraph data can be loaded into Spark DataFrames in various forms:
 
 #### Typed Triples
 
-You can load the entire Dgraph database as triples into a [Apache Spark Dataset]():
+You can load the entire Dgraph database as triples into an [Apache Spark Dataset](https://spark.apache.org/docs/latest/api/java/index.html?org/apache/spark/sql/Dataset.html). For example:
```
[the remark this answers is not in the document]
Comment (EnricoMi) [reply README.md:136]: While you are at it: can you change `Dataset` to `DataFrame`?

File: README.md [hunk 2]
```diff
@@ -399,15 +400,15 @@ in the `pom.xml`file. Otherwise, we would see this error:
...
+Furthermore, we need to set `protobuf-java >= 3.0.0` in the `pom.xml` file:
 
     <dependency>
       <groupId>com.google.protobuf</groupId>
       <artifactId>protobuf-java</artifactId>
       <version>[3,]</version>
     </dependency>
 
-to get rid of this error
+...to get rid of this error:
```
Comment (EnricoMi) [inline README.md:411]: Should there be a whitespace after the ellipsis? Can we use the ellipsis sign `…`?

File: README.md [hunk 3]
```diff
@@ -1,14 +1,14 @@
 # Spark Dgraph Connector
 
-This projects provides an [Apache Spark](https://spark.apache.org/) connector
+This project provides an [Apache Spark](https://spark.apache.org/) connector
```
Comment (EnricoMi) [inline README.md:3]: I have read this bit a thousand times, annoying!

**Comment by EnricoMi** [comment]
> (@EnricoMi Should I raise a PR against the 2.4 branch too, or do you periodically merge things back anyway?)

I will deal with the 2.4 merge, no worries!

**Comment by EnricoMi** [comment]
What do you think about the API? Since you have read the examples in the README.md.

File: README.md [hunk 4]
```diff
@@ -1,14 +1,14 @@
...
 for [Dgraph databases](https://dgraph.io/).
-It comes with [Spark Data Sources](https://spark.apache.org/docs/latest/sql-data-sources.html)
+It comes with a [Spark Data Source](https://spark.apache.org/docs/latest/sql-data-sources.html)
```
[the remark this answers is not in the document]
Comment (herebebeasties) [reply README.md:5]: Looking at their docs, I think "data source" is the term for a provider of data, as opposed to the individual instances of it => singular here?

[the remark this answers is not in the document]
Comment (EnricoMi) [reply README.md:5]: fine with me

File: README.md (hunk 2 above)
Comment (EnricoMi) [reply README.md:411]: nope, fine with me, as long as you use the single character unicode ellipsis :-D

**Pull request merged**